pkg/ansible/controller/reconcile.go

```diff
@@ -189,15 +190,23 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		return reconcileResult, nil
 	}
 	if r.ManageStatus {
-		err = r.markDone(u, statusEvent, failureMessages)
+		err = r.markDone(u, request.NamespacedName, statusEvent, failureMessages)
 		if err != nil {
 			logger.Error(err, "failed to mark status done")
 		}
 	}
 	return reconcileResult, err
 }
 
-func (r *AnsibleOperatorReconciler) markRunning(u *unstructured.Unstructured) error {
+func (r *AnsibleOperatorReconciler) markRunning(u *unstructured.Unstructured, namespacedName types.NamespacedName) error {
+	// Get the latest resource to prevent updating a stale status
+	err := r.Client.Get(context.TODO(), namespacedName, u)
+	if apierrors.IsNotFound(err) {
+		return err
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
Isn't this:
```Go
	if apierrors.IsNotFound(err) {
		return err
	}
	if err != nil {
		return err
	}
```
equivalent to:
```Go
	if err != nil {
		return err
	}
```

Reply:
Thanks for catching that, initially had a log statement there.